Examples/src/main/java/com/mapbox/vision/examples/ExternalCameraSource.java

```diff
@@ -0,0 +1,175 @@
+
+        @Override
+        public void onDetectionsUpdated(@NotNull FrameDetections frameDetections) {
+            visionView.setDetections(frameDetections);
+        }
+
+        @Override
+        public void onSignsUpdated(@NotNull FrameSigns frameSigns) {}
+
+        @Override
+        public void onRoadUpdated(@NotNull RoadDescription roadDescription) {}
+
+        @Override
+        public void onWorldUpdated(@NotNull WorldDescription worldDescription) {}
+
+        @Override
+        public void onVehicleLocationUpdated(@NotNull VehicleLocation vehicleLocation) {}
+
+        @Override
+        public void onCameraUpdated(@NotNull Camera camera) {}
+
+        @Override
+        public void onCountryUpdated(@NotNull Country country) {}
+
+        @Override
+        public void onClientUpdate() {}
+    };
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        visionView = findViewById(R.id.vision_view);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        VisionManager.create(
+                customVideoSource,
+                visionEventsListener
+        );
+        VisionManager.start();
+        VisionManager.setVideoSourceListener(visionView);
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+
+        VisionManager.stop();
+        VisionManager.destroy();
+    }
+
+    /**
+     * Decodes video source frame by frame and feeds frames to Vision SDK.
+     */
+    private void startFileVideoSource() {
+        // Use MediaMetadataRetriever to decode video.
+        // It isn't the fastest approach to decode videos and you probably want some other method
+        // if FPS is important (eg. MediaCodec).
+        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
+        retriever.setDataSource(new File(getExternalFilesDir(""), VIDEO_FILE).getAbsolutePath());
+
+        // Get video frame size.
+        int frameWidth = Integer.parseInt(retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH));
+        int frameHeight = Integer.parseInt(retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT));
+        ImageSize imageSize = new ImageSize(frameWidth, frameHeight);
+        // ByteBuffer to hold RGBA bytes.
+        ByteBuffer rgbaByteBuffer = ByteBuffer.wrap(new byte[frameWidth * frameHeight * 4]);
+
+        // Get duration.
+        long duration = Long.parseLong(retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION));
+
+        try {
+            // Get frames one by one with 1 second intervals.
+            for (int seconds = 0; seconds < duration; seconds++) {
+                Bitmap bitmap = retriever
+                        .getFrameAtTime(seconds * 1000000, MediaMetadataRetriever.OPTION_CLOSEST)
```

Comment:
does `getFrameAtTime` require time in micro-seconds?

Reply:
Yup

Reply:
Bitmap bitmap = retriever.getFrameAtTime(
TimeUnit.SECONDS.toMicros(seconds), 
MediaMetadataRetriever.OPTION_CLOSEST
)

Reply:
Ok